spring-cloud-data-shell/src/test/java/org/springframework/cloud/data/shell/AbstractShellIntegrationTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.shell;
+
+import java.io.IOException;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.rules.TestName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.cloud.data.rest.AdminApplication;
+import org.springframework.cloud.data.shell.command.StreamCommandTemplate;
+import org.springframework.shell.Bootstrap;
+import org.springframework.shell.core.CommandResult;
+import org.springframework.shell.core.JLineShellComponent;
+import org.springframework.util.AlternativeJdkIdGenerator;
+import org.springframework.util.Assert;
+import org.springframework.util.IdGenerator;
+import org.springframework.util.SocketUtils;
+
+/**
+ * Base class for all the Shell integration test classes.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public abstract class AbstractShellIntegrationTest {
+
+	private static final Logger logger = LoggerFactory.getLogger(AbstractShellIntegrationTest.class);
+
+	private final IdGenerator idGenerator = new AlternativeJdkIdGenerator();
+
+	/**
+	 * This property can be set to false when running test suite.
+	 */
+	private static final boolean SHUTDOWN_AFTER_RUN = true;
+
+	private static AdminApplication application;
+
+	private static JLineShellComponent shell;
+
+	private static CloudDataShell cloudDataShell;
+
+	/**
+	 * Use available TCP port for the admin port
+	 */
+	private static final int adminPort = SocketUtils.findAvailableTcpPort();
+
+	/**
+	 * Used to capture currently executing test method.
+	 */
+	@Rule
+	public TestName name = new TestName();
+
+	@BeforeClass
+	public static synchronized void startUp() throws InterruptedException, IOException {
+		if (application == null) {
+			application = new AdminApplication();
+			application.main(new String[] {
+					String.format("--server.port=%s", adminPort),
+					"--spring.main.show_banner=false"});
+		}
+		shell = new Bootstrap(new String[] {"--port", String.valueOf(adminPort)}).getJLineShellComponent();
```

Comment:
It's worth adding a delegating stop() in the CloudDataShell, thus no need to have both the shell and cloudDataShell fields? (could just have the CloudDataShell assigned to a 'shell' field instead)